server.js

```diff
@@ -49,6 +51,27 @@ server.register({ register: require('./api') }, function(err) {
     });
     throw err;
   }
+
+  server.ext('onPreResponse', function(req, reply) {
+    if (req.response.isBoom) {
+      return reply.continue();
+    }
+
+    // Once a successful request completes, we delete any
+    // temporary files we created
+    req.response.once('finish', function() {
+      if (req.app.tmpFile) {
+        fs.unlink(req.app.tmpFile, function(err) {
+          if (err) {
+            throw 'Failed to destroy temporary file with ' + err;
```

Comment:
Good catch. My real mistake here was not explicitly handling the error, since HAPI will create an error response when you have an unhandled exception in the request lifecycle.